### PR TITLE
Fix cloudbuild and simplify Makefile & cloudbuild

### PR DESCRIPTION
**Pull request by nckturner**
- Remove cloudbuild logic from Makefile
- Fix ldflags version
- Use gcr.io/cloud-builders/docker for cloudbuild
  (tangentially, this should also fix the cloudbuild
  build, which was failing because of what appeared
  to be an env var collision between the Makefile
  and the build container with $IMAGE).

**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>

/kind cleanup

**What this PR does / why we need it**:
Simplifies cloudbuild and Makefile (should fix it in the process, but we will need to merge it to see).  

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,58 +13,26 @@
...
-	DEV_TAG = $(TAG)
-else
-	DEV_TAG = $(GIT_VERSION)
-endif
-IMAGE ?= $(REGISTRY)/$(IMAGE_NAME):$(DEV_TAG)
-
-RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
-RELEASE_IMAGE ?= $(REGISTRY)/$(IMAGE_NAME):$(RELEASE_TAG)
+VERSION ?= $(GIT_VERSION)
+IMAGE := amazon/cloud-controller-manager:$(VERSION)
```
Comment (andrewsykim) [inline Makefile:21]: is this a real image or just a dummy replacement?

File: Dockerfile [hunk 2]
```diff
@@ -32,20 +32,19 @@ ARG DISTROLESS_IMAGE=gcr.io/distroless/static@sha256:c6d5981545ce1406d33e61434c6
...
 
 WORKDIR /build
 COPY go.mod go.sum ./
 COPY cmd/ cmd/
 COPY pkg/ pkg/
 COPY Makefile ./Makefile
-ENV CGO_ENABLED=0
-ENV GOPROXY ${GOPROXY:-https://proxy.golang.org}
-RUN make aws-cloud-controller-manager
+RUN GO111MODULE=on CGO_ENABLED=0 GOOS=${GOOS} GOPROXY=${GOPROXY} go build \
```
Comment (ayberk) [inline Dockerfile:44]: Any specific reason why we're not simply running make anymore?

File: Makefile (hunk 1 above)
Comment (nckturner) [reply Makefile:21]: Dummy replacement for now, but would be replaced by something real.

File: Dockerfile (hunk 2 above)
Comment (nckturner) [reply Dockerfile:44]: I didn't like the circular dependency of make -> dockerfile -> make.  It makes the env var/docker args logic hard to follow IMO.

File: Makefile (hunk 1 above)
Comment (nckturner) [reply Makefile:21]: Well to be clear, I was really intending this to just be the name we gave it so that it makes sense locally when listed with `docker images`, and the official image logic for gcr, ecr (when we add it) would be explicit, e.g. the the gcr logic is entirely contained in the cloudbuild.yaml file.

File: Dockerfile (hunk 2 above)
Comment (ayberk) [reply Dockerfile:44]: I see. I was more concerned about changing the build command on Makefile and then forgetting to update it here. I'm not super familiar with the best practices here, so I'll leave it to your judgement.

Comment (nckturner) [reply Dockerfile:44]: Yeah I considered the downside but I don't think the command will change much at all

File: Dockerfile [hunk 3]
```diff
@@ -32,20 +32,19 @@ ARG DISTROLESS_IMAGE=gcr.io/distroless/static@sha256:c6d5981545ce1406d33e61434c6
...
-ARG GOPROXY
+ARG VERSION
+ARG GOPROXY=https://goproxy.io,direct
+ARG GOOS=linux
 
 WORKDIR /build
 COPY go.mod go.sum ./
 COPY cmd/ cmd/
 COPY pkg/ pkg/
 COPY Makefile ./Makefile
```
Comment (wongma7) [inline Dockerfile:43]: dont need to copy anymore

File: cloudbuild.yaml [hunk 4]
```diff
@@ -1,19 +1,16 @@
...
-    - release-staging
+    - build
+    - --tag=gcr.io/$PROJECT_ID/k8s-staging-provider-aws:$_GIT_TAG
+    - --tag=gcr.io/$PROJECT_ID/k8s-staging-provider-aws:latest
+    - --build-arg=VERSION=$_GIT_TAG \
+    - .
 substitutions:
-  # _GIT_TAG will be filled with a git-based tag for the image, of the form vYYYYMMDD-hash, and
-  # can be used as a substitution
   _GIT_TAG: '12345'
```
Comment (wongma7) [inline cloudbuild.yaml:15]: is this intentional

**Review by wongma7 — APPROVED** [review]
